Add vulnerable option to dotnet list package doc

ciwchris opened the pull request:
## Summary

The `dotnet list package` package is out of date, it is missing the `vulnerable` option. I used the output from `dotnet list package --help` as the content of my proposed changes.

Feel free to close this PR and update the documentation through a different PR if there's another preferred workflow for updating this content.

ciwchris committed:
Add vulnerable option to dotnet list package doc

tdykstra commented on docs/core/tools/dotnet-list-package.md:
````diff
@@ -15,7 +15,7 @@ ms.date: 11/11/2020
 ```dotnetcli
 dotnet list [<PROJECT>|<SOLUTION>] package [--config <SOURCE>]
-    [--deprecated]
+    [--deprecated] [--vulnerable]
     [--framework <FRAMEWORK>] [--highest-minor] [--highest-patch]
     [--include-prerelease] [--include-transitive] [--interactive]
     [--outdated] [--source <SOURCE>] [-v|--verbosity <LEVEL>]
````
We generally try to present the options in alphabetic order.

```suggestion
    [--deprecated]
    [--framework <FRAMEWORK>] [--highest-minor] [--highest-patch]
    [--include-prerelease] [--include-transitive] [--interactive]
    [--outdated] [--source <SOURCE>] [-v|--verbosity <LEVEL>]
    [--vulnerable]
```

tdykstra commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -79,6 +79,10 @@ The project or solution file to operate on. If not specified, the command search
 
   Displays packages that have been deprecated.
 
+- **`--vulnerable`**
+
+  Lists packages that have known vulnerabilities. Cannot be combined with `--deprecated` or `--outdated` options.
+  
```
This should go at the end of the list, following --verbosity.

tdykstra reviewed (changes requested):
@ciwchris Thanks for this submission! Just a couple minor adjustments and it's ready to merge.

ciwchris committed:
Place vulnerable option in alphabetical order

The list of options for the dotnet tool is in alphabetical order. Therefore the newly added `vulnerable` option needs to be moved to the end of the list so that it appears in alphabetical order.

tdykstra reviewed (approved):
Looks good, thanks!

ciwchris commented:
@tdykstra sorry about the extra space though. 😞 I didn't want to make another commit to remove it. I was going to clone the repo down so I could amend the last commit but it's a big repo and was taking a long time. So then I was going to try to clone just the last so many commits so I could quickly amend but then needed to take care of something else for a bit.

tdykstra commented:
Not a problem at all, thanks again for catching and correcting the omission of --vulnerable!
